python/lsst/ts/pmd/mock_server.py

```diff
@@ -0,0 +1,98 @@
+import logging
+import queue
+import math
+
+import serial
+
+
+class MockSerial(serial.Serial):
```

Comment:
I'm curious: do you gain any useful behavior from serial.Serial? I'm a bit worried that you might gain unwanted behavior -- e.g. accidentally talking to a real serial port.

If you need the behavior and want to be safe you could force port to be some safe port (is /dev/null usable as a serial port?) -- either by checking the port argument or ignoring it. Ignoring it is most flexible, but risks errors if using introspection.

Reply:
```
def connect(self):
        """Connect to the device."""
        if not self.simulation_mode:
            self.commander = serial.Serial(port=SIMULATION_SERIAL_PORT)
        else:
            main, reader = pty.openpty()
            self.commander = MockSerial(os.ttyname(main))
```
The connect function opens a pseudo teletype if in simulation mode, so it is gaining some useful behavior opening the "port".

Reply:
That sounds good. Thank you for the explanation.